### PR TITLE
Fix rescheduling error with attendee phone number as location

### LINKED ISSUE #5652: [CAL-437] Rescheduling a meeting with attendee's phone number as a location throws an error
[https://app.birdeatsbug.com/sessions/62H9W7sOUwthUMkkVnRd4oBWJUGIoQUYTw5BB3XrKOM8](https://app.birdeatsbug.com/sessions/62H9W7sOUwthUMkkVnRd4oBWJUGIoQUYTw5BB3XrKOM8)

Created via Threads. See full discussion: [https://threads.com/34435195648](https://threads.com/34435195648)

> From [SyncLinear.com](https://synclinear.com) | [CAL-437](https://linear.app/calcom/issue/CAL-437/rescheduling-a-meeting-with-attendees-phone-number-as-a-location)

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Fixes rescheduling error for bookings that have the attendee's phone number as their location. 

Also fixes that SMS reminder phone number was always shown on success page when the location was attendee's phone number. 

Fixes #5652

**Environment**: Staging(main branch)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Book an event with attendee's phone number as location
- Reschedule the event

### CODE REVIEW

File: apps/web/components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -622,7 +622,7 @@ const BookingPage = ({
                   )}
                 </>
                 {/* TODO: Change name and id ="phone" to something generic */}
-                {AttendeeInput && (
+                {AttendeeInput && !disableInput && (
```
Comment (CarinaWolli) [inline apps/web/components/booking/pages/BookingPage.tsx:625]: Don't show the disabled input field. We anyways already show the selected location above

File: apps/web/pages/success.tsx [hunk 2]
```diff
@@ -452,7 +458,7 @@ export default function Success(props: SuccessProps) {
                           </>
                         );
                       })}
-                    {bookingInfo?.smsReminderNumber && (
+                    {bookingInfo?.smsReminderNumber && hasSMSAttendeeAction && (
```
Comment (CarinaWolli) [inline apps/web/pages/success.tsx:461]: Only show sms reminder number on success page if the event type of the booking actually has a workflow step with the SMS_ATTENDEE action

File: packages/prisma/zod-utils.ts [hunk 3]
```diff
@@ -153,7 +153,7 @@ export const extendedBookingCreateBody = bookingCreateBodySchema.merge(
     allRecurringDates: z.string().array().optional(),
     currentRecurringIndex: z.number().optional(),
     rescheduleReason: z.string().optional(),
-    smsReminderNumber: z.string().optional(),
+    smsReminderNumber: z.string().optional().nullable(),
```
Comment (CarinaWolli) [inline packages/prisma/zod-utils.ts:156]: This fixes the rescheduling error

File: apps/web/pages/success.tsx [hunk 4]
```diff
@@ -298,6 +298,11 @@ export default function Success(props: SuccessProps) {
 
   const locationToDisplay = getSuccessPageLocationMessage(location, t);
 
+  const hasSMSAttendeeAction =
+    eventType.workflows.find((workflowEventType) =>
+      workflowEventType.workflow.steps.find((step) => step.action === WorkflowActions.SMS_ATTENDEE)
+    ) !== undefined;
```
Comment (alannnc) [inline apps/web/pages/success.tsx:304]: Here we will only run till we found our SMS_ATTENDEE action.

**Review by alannnc — APPROVED** [review]

**Pull request merged**